### PR TITLE
fix: Improve the experience for users who need to do a checkpoint upgrade

### PR DESCRIPTION
**Pull request by harshilp24**

### CODE REVIEW

File: website/docs/getting-started/setup/installation-guides/docker/README.mdx [hunk 1]
```diff
@@ -81,7 +81,8 @@ The Appsmith server should be up and running soon and can be accessed at [http:/
 
 ### Updating Appsmith
 :::caution
-   It's recommended to backup the Appsmith instance before performing an update. For more information, see [How to create a backup](https://docs.appsmith.com/getting-started/setup/instance-management/appsmithctl#backup-appsmith-instance).
+   * It's recommended to backup the Appsmith instance before performing an update. For more information, see [How to create a backup](https://docs.appsmith.com/getting-started/setup/instance-management/appsmithctl#backup-appsmith-instance).
+* When you are manually updating the Appsmith instance, ensure that you [upgrade to the checkpoint](/getting-started/setup/instance-management#checkpoint-upgrade) version `v1.9.2` released between your version and the latest version.
```
Comment (sum35h) [inline website/docs/getting-started/setup/installation-guides/docker/README.mdx:85]: Let's specify saying, if and only if your current `Appsmith version < v1.9.2` then follow the instructions to upgrade to checkpoint.

**Review by sum35h — CHANGES_REQUESTED** [review]

File: website/docs/getting-started/setup/installation-guides/docker/README.mdx [hunk 2]
```diff
@@ -81,9 +81,12 @@ The Appsmith server should be up and running soon and can be accessed at [http:/
 
 ### Updating Appsmith
 :::caution
-   It's recommended to backup the Appsmith instance before performing an update. For more information, see [How to create a backup](https://docs.appsmith.com/getting-started/setup/instance-management/appsmithctl#backup-appsmith-instance).
+   * It's recommended to backup the Appsmith instance before performing an update. For more information, see [How to create a backup](https://docs.appsmith.com/getting-started/setup/instance-management/appsmithctl#backup-appsmith-instance).
+* If you are manually updating the Appsmith instance and your current version of Appsmith is older than `v1.9.2`,  make sure that you [upgrade to the checkpoint](/getting-started/setup/instance-management#checkpoint-upgrade) version `v1.9.2` released between your version and the latest version.
```
Comment (sum35h) [inline website/docs/getting-started/setup/installation-guides/docker/README.mdx:85]:
```suggestion
* If you are manually updating the Appsmith instance and your current version of Appsmith is less than `v1.9.2`,  make sure that you [upgrade to the checkpoint](/getting-started/setup/instance-management#checkpoint-upgrade) version `v1.9.2` released between your version and the latest version.
```

File: website/docs/getting-started/setup/instance-management/README.mdx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+</Tabs>
+
+Save this file, then run the following command on your shell to restart Appsmith.
+
+```
+docker-compose up -d
+```
+
+:::note
+In this case, version `1.9.2` was used. If required, you can also use a specific older version.
```
Comment (sum35h) [inline website/docs/getting-started/setup/instance-management/README.mdx:81]: I don't think this makes sense, currently, our only mandatory checkpoint is v1.9.2.
Users have to hit this checkpoint migration if they are going to >1.9.2, not when going back to <1.9.2

File: website/docs/getting-started/setup/instance-management/README.mdx [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+:::
```
Comment (sum35h) [inline website/docs/getting-started/setup/instance-management/README.mdx:82]:
```suggestion
```

**Review by sum35h — CHANGES_REQUESTED** [review]

**Review by sum35h — APPROVED** [review]

File: website/docs/help-and-support/troubleshooting-guide/deployment-errors.mdx [hunk 5]
```diff
@@ -135,98 +135,6 @@ As the error message in the server logs suggests, your Appsmith instance has mis
...
-
-Run the restart command once again: 
-
- ```
- docker-compose up -d
-  ```
-:::note 
-If you haven't been using your Appsmith instance for a while, it could be possible that there might be multiple versions that show up in the error message. In such a case, perform the same steps for each version in order, starting from the oldest.
-:::
+To resolve this error, you may need to manually map your instance to the required version, in this case, ```v1.9.2```. You can check this [Checkpoint upgrade guide](/getting-started/setup/instance-management#checkpoint-upgrade) to learn more about checkpoint upgrade.
```
Comment (danciaclara) [inline website/docs/help-and-support/troubleshooting-guide/deployment-errors.mdx:138]: Change this to 

> To resolve this error, you must [upgrade to the checkpoint version](/getting-started/setup/instance-management#checkpoint-upgrade), in this case, v1.9.2.

File: website/docs/getting-started/setup/instance-management/README.mdx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+```bash
+// To restart appsmith without docker compose
+docker pull appsmith/appsmith-ce && docker restart appsmith
+
+// To restart appsmith with docker compose
+docker-compose pull && docker-compose up -d --force-recreate appsmith
+```
+
+## Checkpoint upgrade
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-management/README.mdx:22]: Checkpoint version and upgrades

File: website/docs/getting-started/setup/instance-management/README.mdx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+Upgrading a Docker container checkpoint to the latest version is a crucial step in maintaining the stability, security, and performance of the containerized applications.
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-management/README.mdx:23]: Replace with 

> When you upgrade manually, you must upgrade to the checkpoint version released between your version and the latest version.

File: website/docs/getting-started/setup/instance-management/README.mdx [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+docker pull appsmith/appsmith-ce && docker restart appsmith
+
+// To restart appsmith with docker compose
+docker-compose pull && docker-compose up -d --force-recreate appsmith
+```
+
+## Checkpoint upgrade
+Upgrading a Docker container checkpoint to the latest version is a crucial step in maintaining the stability, security, and performance of the containerized applications. 
+
+You may need to manually map your instance to the required version If your current version of Appsmith is less than `v1.9.2`. To do this, access your Appsmith instance and make the following change to the `docker-compose.yml` file:
```
Comment (danciaclara) [inline website/docs/getting-started/setup/instance-management/README.mdx:25]: Replace with 

**Appsmith checkpoint versions**
         - v1.9.2

Access your Appsmith instance and make the following change to the `docker-compose.yml` file:

File: website/docs/getting-started/setup/installation-guides/docker/README.mdx [hunk 9]
```diff
@@ -81,9 +81,12 @@ The Appsmith server should be up and running soon and can be accessed at [http:/
 
 ### Updating Appsmith
 :::caution
-   It's recommended to backup the Appsmith instance before performing an update. For more information, see [How to create a backup](https://docs.appsmith.com/getting-started/setup/instance-management/appsmithctl#backup-appsmith-instance).
+   * It's recommended to backup the Appsmith instance before performing an update. For more information, see [How to create a backup](https://docs.appsmith.com/getting-started/setup/instance-management/appsmithctl#backup-appsmith-instance).
```
Comment (danciaclara) [inline website/docs/getting-started/setup/installation-guides/docker/README.mdx:84]: Replace with below

-  [Create a backup](https://docs.appsmith.com/getting-started/setup/instance-management/appsmithctl#backup-appsmith-instance) of the Appsmith instance before performing a manual update.
- If your version is less than **v1.9.2**, [upgrade to the checkpoint version](https://docs.appsmith.com/getting-started/setup/instance-management#checkpoint-upgrade) v1.9.2 before upgrading to the latest version.

**Review by danciaclara — COMMENTED** [review]
@harshilp24 Can you create a separate PR and quickly make these updates please?

**Pull request merged**